src/agent/src/rpc.rs

```diff
@@ -319,23 +312,26 @@ impl AgentService {
 
         let pipe_size = AGENT_CONFIG.container_pipe_size;
 
-        let p = if let Some(p) = oci.process() {
-            #[cfg(feature = "guest-pull")]
-            {
-                let new_p = image::get_process(p, &oci, req.storages.clone())?;
-                Process::new(&sl(), &new_p, cid.as_str(), true, pipe_size, proc_io)?
-            }
-
-            #[cfg(not(feature = "guest-pull"))]
-            Process::new(&sl(), p, cid.as_str(), true, pipe_size, proc_io)?
-        } else {
+        let Some(p) = oci.process() else {
             info!(sl(), "no process configurations!");
             return Err(anyhow!(nix::Error::EINVAL));
         };
 
+        let process = match confidential_data_hub::is_cdh_client_initialized() {
+            true => {
+                let new_p = crate::confidential_data_hub::image::get_process(
+                    p,
+                    &oci,
+                    req.storages.clone(),
+                )?;
+                Process::new(&sl(), &new_p, cid.as_str(), true, pipe_size, proc_io)?
```

Comment:
Why does this depend on the CDH client being initialized? If I wanted to use the CDH client but without guest pull, would that still work?

Overall, would it be possible to just always use an in-VM pause bundle?

Reply:
> Why does this depend on the CDH client being initialized? If I wanted to use the CDH client but without guest pull, would that still work?

It still work. The logic inside `crate::confidential_data_hub::image::get_process(` will detect if any image pull volume in `req.storages` is injected. If so, will get pause image process.

> Overall, would it be possible to just always use an in-VM pause bundle?

Sorry, I'm not an expert here. I'm trying to retain the original logic here. Maybe this problem needs to be solved by a separate issue or PR?